File: Models/Movie.cs
using System.ComponentModel.DataAnnotations;

namespace Models
{
    using System.Text.Json.Serialization;

    public class Movie
    {
        [JsonPropertyName("id")]
        public int Id { get; set; }

        [JsonPropertyName("title")]
        public string MovieTitle { get; set; }

        [JsonPropertyName("overview")]
        public string Overview { get; set; }

        [JsonPropertyName("original_language")]
        public string Language { get; set; }

        [JsonPropertyName("poster_path")]
        public string PosterPath { get; set; }

        [JsonPropertyName("release_date")]
        public string ReleaseDate { get; set; }

        [JsonPropertyName("vote_average")]
        public double Rating { get; set; }

        [JsonPropertyName("budget")]
        public int Budget { get; set; }
        [JsonPropertyName("revenue")]
        public int Revenue { get; set; }

        [JsonPropertyName("runtime")]
        public int RunTime { get; set; }

        [JsonPropertyName("Genres")]
        public List<Genre> Genres { get; set; } = new List<Genre>();

        [JsonPropertyName("production_companies")]
        public List<Production> ProductionCompanies { get; set; } = new List<Production>();

        [JsonPropertyName("backdrops")]
        public List<Images> Backdrops { get; set; } = new List<Images>();

        [JsonPropertyName("videos")]
        public List<Videos> Videos { get; set; } = new List<Videos>();

        public string FullPosterUrl =>
            !string.IsNullOrWhiteSpace(PosterPath)
                ? $"https://image.tmdb.org/t/p/w500{PosterPath}"
                : "/images/placeholder.png";
    }

}

File: frontend/moviedatabase/lib/Models/Movie.dart
class Movie {
  final int id;
  final String title;
  final String overview;
  final String posterPath;
  final String releaseDate;
  final double voteAverage;
  final String originalLanguage;
  final int? budget;
  final int? revenue;
  final int? runtime;

  Movie({
    required this.id,
    required this.title,
    required this.overview,
    required this.posterPath,
    required this.releaseDate,
    this.voteAverage = 0.0,
    this.originalLanguage = '',
    this.budget,
    this.revenue,
    this.runtime,
  });

  factory Movie.fromJson(Map<String, dynamic> json) {
    return Movie(
      id: json['id'] ?? 0,
      title: json['title'] ?? json['movieTitle'] ?? 'Unknown Title',
      overview: json['overview'] ?? 'No overview available',
      posterPath: json['poster_path'] ?? json['posterPath'] ?? '',
      releaseDate: json['release_date'] ?? json['releaseDate'] ?? '',
      voteAverage: (json['vote_average'] ?? json['rating'] ?? 0.0).toDouble(),
      originalLanguage: json['original_language'] ?? json['language'] ?? '',
      budget: json['budget'],
      revenue: json['revenue'],
      runtime: json['runtime'] ?? json['runTime'],
    );
  }

  String get fullPosterUrl {
    if (posterPath.isEmpty) return '';
    return posterPath.startsWith('http')
        ? posterPath
        : 'https://image.tmdb.org/t/p/w500$posterPath';
  }
}
